Bio/Entrez/__init__.py

```diff
@@ -502,19 +505,28 @@ def _open(cgi, params=None, post=None, ecitmatch=False):
     This function also enforces the "up to three queries per second rule"
     to avoid abusing the NCBI servers.
     """
-    # NCBI requirement: At most three queries per second.
+    # NCBI requirement: At most three queries per second if no API key is provided.
     # Equivalently, at least a third of second between queries
-    delay = 0.333333334
-    current = time.time()
-    wait = _open.previous + delay - current
-    if wait > 0:
-        time.sleep(wait)
-        _open.previous = current + wait
-    else:
-        _open.previous = current
-
     params = _construct_params(params)
     options = _encode_options(ecitmatch, params)
+    if params['api_key'] is None:
```

Comment:
Two point here.

First, style wise ``if params['api_key']:`` feels more Pythonic to me (and means swapping the if/else clauses round).

Second, other than checking the delay, isn't all of the if/else code identical?

Combined, you could write this as:

```python
delay = 0.1 if params['api_key'] else 0.333333334
```

(and leave the rest of the code as is)